package.json

```
@@ -52,7 +52,8 @@
         "build": "tsc && vite build",
         "format": "prettier --write .",
         "start": "vite build --watch",
-        "test": "[ -f ./dist/index.js ] || npm run build && vitest",
+        "test": "[ -f ./dist/index.js ] || npm run test:build",
+        "test:build": "npm run build && vitest",
```

This isn't doing quite what you expect - previously it was running the build if it hadn't run already - we actually don't need that conditional run, so can just have `vitest` on its own - I was mixing up `vite preview` and `vitest`!

```suggestion
        "start:test": "vitest",
        "test": "vitest run",
```

---

`run build` is required to capture changes to the codebase because tests run specifically against the files in /dist, and without it the test system will only reflect changes to the tests themselves. `test:build` forces a build of the codebase to ensure all tests run successfully against the latest code.

if there is another way to capture code changes in list I would love to use it because this honestly isn't great.

---

Good point - it still needs that ternary for the moment then - I'm about to be pushing a test update that tests the built cjs (currently) via the new commandline tool and a bash script (the intention is that other implementations can support the same arguments and then run the same tests) - currently I can't get nodejs to load binary data properly though, so raw and uint8array both fail to decode :-(

---

All the test code has been updated and fixed, so it works properly - the test strategy will be changing slightly for the production code so it'll only run when it can (and force it on github itself) - so I'll clean up this when I get to it - can undo the change in the meantime :-)

---

This can now be removed - the test stategy has been updated (including better tests for built code, and an external test tool [here](https://github.com/pieroxy/lz-string/tree/master/test) - I'll get it running in github actions soon!)